### PR TITLE
Partially implement XML/XMLList abstract equality

### PR DESCRIPTION
**Pull request by evilpie**
Helps #10474 and  #10183. Quite likely also others, previously the `==` comparison would return wrong results which is often not an obvious failure.

### CODE REVIEW

File: core/src/avm2/value.rs [hunk 1]
```diff
@@ -1138,6 +1138,31 @@ impl<'gc> Value<'gc> {
         other: &Value<'gc>,
         activation: &mut Activation<'_, 'gc>,
     ) -> Result<bool, Error<'gc>> {
+        // ECMA-357 extends the abstract equality algorithm with steps
+        // for XML and XMLList types. Because they are objects in Ruffle we
+        // have to do be a bit more complicated and factor out the code into
```
Comment (Herschel) [inline core/src/avm2/value.rs:1143]: typo: `do be` -> `be`

**Comment by n0samu** [comment]
Can confirm this makes #10183 playable! 🎉

File: core/src/avm2/value.rs (hunk 1 above)
Comment (evilpie) [reply core/src/avm2/value.rs]: Thanks!

**Review by Herschel — APPROVED** [review]
Thank you!

**Comment by Lord-McSweeney** [comment]
Unfortunately this seems to have regressed #10390. The menu doesn't load anymore.

**Pull request merged**